### DISCUSSION TITLE
Auto Refresh token is not working

### DISCUSSION THREAD

**Discussion by LeonardoMussato** [nuxt-community/auth-module#1575] [Support Requests]
Hello to everyone, I set up my refresh scheme as the official docs said but I still have issues when my token expires... 

For example when my token expire I would like to have my token refreshed without the need of a new login.

Is it possible to do it?

Thanks in advanced for your time.

**Comment by skoenfaelt** [comment]
Connecting https://github.com/nuxt-community/auth-module/discussions/1694

**Comment by sneakylenny** [comment]
Can you send us a example of the token issued? 
My suspicions are the token doesn't contain a "exp" (expiry timestamp) in the payload.
If you can't provide this you can set the maxAge attribute in the config like so:

```js
auth: {
  // ...
  strategies: {
    local: {
      scheme: 'refresh',
      tokenType: 'JWT',
      token: {
        property: 'token',
        maxAge: 60 * 60, // 60 Minutes
        global: true,
      },
      refreshToken: {
        tokenRequired: true,
        property: 'refreshToken',
        data: 'refreshToken',
        maxAge: 60 * 60 * 24 * 30 // 30 Days
      },
      // ...
    }
  }
  // ...
}
```

**Comment by sneakylenny** [comment]
Also make sure the [autoLogout](https://auth.nuxtjs.org/schemes/refresh#autologout) property is set to **false**
> If the token has expired, it will prevent the token from being refreshed on load the page and force logout the user.

**Comment by skoenfaelt** [comment]
does not work, even if the max age is far lower than the actual exp of the token.

**Reply by sneakylenny** [reply]
`maxAge` is merely a fallback in case you token doesn't contain a `exp` prop.
The token's `exp` it prioritized over the `maxAge` prop:

![Screenshot_220](https://user-images.githubusercontent.com/60601502/168294136-74d6cd61-eb98-4bb4-8e5c-b0c54efe9398.png)
